src/Pizzeria.cpp

```diff
@@ -3,16 +3,17 @@
 #include <algorithm>
 #include <thread>
 #include <iostream>
+#include <memory>
 
-Pizzeria::Pizzeria(std::string const & name)
-    : name_(name)
-    , orders_()
+
+Pizzeria::Pizzeria(std::string const & name, std::unique_ptr<ITimer> timer)
+    : name_(name) , orders_(), timer_ptr(std::move(timer))
 {}
 
 int Pizzeria::makeOrder(Pizzas pizzas)
 {
     int orderId = rand() % 1000;
-    orders_.push_back(std::make_tuple(orderId, pizzas, std::chrono::system_clock::now(), Status::New));
+    orders_.push_back(std::make_tuple(orderId, pizzas, timer_.now(), Status::New));
```

Comment:
timer_? or timer_ptr? Why do we have them both? We should use only one of such objects.